src/System.Security.Cryptography.X509Certificates/ref/System.Security.Cryptography.X509Certificates.cs

```diff
@@ -501,6 +531,16 @@ public enum X509RevocationMode
         Offline = 2,
         Online = 1,
     }
+    public abstract partial class X509SignatureGenerator
+    {
+        protected X509SignatureGenerator() { }
+        public System.Security.Cryptography.X509Certificates.PublicKey PublicKey { get { throw null; } }
+        protected abstract System.Security.Cryptography.X509Certificates.PublicKey BuildPublicKey();
+        public static System.Security.Cryptography.X509Certificates.X509SignatureGenerator CreateForECDsa(System.Security.Cryptography.ECDsa key) { throw null; }
```

Comment:
Maybe a better version of wrapping them would be when someone adds a custom generator for a new signature type.  They'd be able to use the existing GetPublicKey member, and just have to write their own AlgorithmIdentifier and (maybe) Signature methods.